File: BackEnd/TemplaRestaurant/src/main/java/Templa/Tesis/App/dtos/ReporteClientesReservasDTO.java
package Templa.Tesis.App.dtos;

import Templa.Tesis.App.Enums.EventoReserva;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class ReporteClientesReservasDTO {
    private String nombreCompleto;
    private String email;
    private String telefono;
    private Long totalReservas;
    private EventoReserva eventoMasFrecuente;
}

File: TemplaFront/src/app/componentes/models/ReporteClientesReservasDTO.ts
import { EventoReserva } from './EventoReserva';

export interface ReporteClientesReservasDTO {
  nombreCompleto: string;
  email: string;
  telefono: string;
  totalReservas: number;
  eventoMasFrecuente: EventoReserva;
}

// ✅ Interface para datos del gráfico de torta
export interface ClienteReservaChartData {
  name: string;
  value: number;
  email: string;
  telefono: string;
  eventoFrecuente: string;
}